packages/api/src/command/medical/document/document-bulk-download.ts

```diff
@@ -0,0 +1,170 @@
+import { Config } from "../../../shared/config";
+import { searchDocuments } from "../../../external/fhir/document/search-documents";
+import { chunk } from "lodash";
+import { processPatientDocumentRequest, MAPIWebhookStatus } from "./document-webhook";
+import {
+  DocumentBulkDownloadDTO,
+  toDTO,
+} from "../../../routes/medical/dtos/document-bulk-downloadDTO";
+
+import {
+  DocumentBulkDownloadProgress,
+  isDocBulkDownloadProcessing,
+  DocumentDownloadStatus,
+} from "../../../domain/medical/document-bulk-download";
+//import { getSignedUrls } from "@metriport/core/external/aws/lambda-logic/document-bulk-signing";
+import { Util } from "../../../shared/util";
+import { getPatientOrFail } from "../patient/get-patient";
+import { Patient } from "../../../domain/medical/patient";
+import { uuidv7 } from "@metriport/core/util/uuid-v7";
+import { storeBulkDownloadQueryInit } from "../patient/query-init";
+import { appendDocBulkDownloadProgress } from "../patient/append-bulk-doc-download-progress";
+import { DocumentReference } from "@medplum/fhirtypes";
+import { makeLambdaClient } from "../../../external/aws/lambda";
+import { getLambdaResultPayload } from "@metriport/core/external/aws/lambda";
+// import { getSignedUrls } from "@metriport/core/external/aws/lambda-logic/document-bulk-signing";
+
+const BATCH_SIZE = 100;
```

Comment:
this is just a guess. Hard for me to tell whats optimal here since constraints are not memory of lambda but rather because we restart the API at any time and it has only a few seconds to shut down, interrupting any long-running operation. The other consideration is that in the current setup, we will send one webhook per batch. So 10000 docs at 100 batch size means receiving 100 webhook payloads.